coreapi.go

```diff
@@ -43,6 +43,9 @@ type CoreAPI interface {
 	// PubSub returns an implementation of PubSub API
 	PubSub() PubSubAPI
 
+	// DNS returns an implementation of DNS API
+	DNS() DNSAPI
```

Comment:
@Stebalien do we want to expose a DNS API in the CoreAPI? It seems like it's not really the intention as the CoreAPI is supposed to deal with how an IPNS node functions not just acting as a wrapper around all low level go-ipfs commands.

@vyzo the alternative to using the CoreAPI in your go-ipfs PR would be to just define the function on the IpfsNode and use `cmdenv.GetNode(env, req)` instead of `GetApi`

Reply:
Also note that we can make the DNS api wider, we have a couple of more methods that would be useful to expose -- `LookupIPAddr` for straight up dns resolution and `Resolve` for multiaddr resolution.

Reply:
I really don't think we want to expose this as it has nothing to do with IPFS.